Library/Homebrew/unpack_strategy.rb

```diff
@@ -160,6 +160,21 @@ def extract_nestedly(to: nil, basename: nil, verbose: false, prioritize_extensio
     Dir.mktmpdir("homebrew-unpack", HOMEBREW_TEMP) do |tmp_unpack_dir|
       tmp_unpack_dir = Pathname(tmp_unpack_dir)
 
+      # Make sure files inside the temporary directory have the same group as the brew instance.
+      #
+      # @see https://github.com/Homebrew/brew/blob/4.4.0/Library/Homebrew/mktemp.rb#L57-L72
+      group_id = if HOMEBREW_BREW_FILE.grpowned?
+        HOMEBREW_BREW_FILE.stat.gid
+      else
+        Process.gid
+      end
+      begin
+        tmp_unpack_dir.chown(nil, group_id)
+      rescue Errno::EPERM
+        require "etc"
+        opoo "Failed setting group \"#{Etc.getgrgid(group_id)&.name}\" on #{tmp_unpack_dir}"
```

Comment:
Should better handle here where there's no group name. because `Failed setting group ""` doesn't seem great.

Reply:
In theory this should never happen. Could switch to `T.must` like other code or add `|| group_id` but code is unreachable.

`Etc.getgrgid` will throw an exception on non-existent id (which also shouldn't happen?).

Reply:
The only scenario that `Etc.getgrgid` returns nil is if you compile Ruby on a system without `getgrent` function.

So I guess Windows maybe? Though `chown` is no-op there so `Errno::EPERM` wouldn't even throw.

Reply:
Also note what I did here: https://github.com/Homebrew/brew/blob/d1e539cb8424d578e4b633f26c3cd31f9851c63e/Library/Homebrew/utils/github/api.rb#L139-L145.

I guess the misconfigured NSS could apply here too?

Reply:
How about:
```suggestion
        group_name = begin
          Etc.getgrgid(group_id)&.name
        rescue ArgumentError
          # Cover for misconfigured NSS setups
          nil
        end
        opoo "Failed setting group \"#{group_name || group_id}\" on #{tmp_unpack_dir}"
```

Reply:
LGTM